beetsplug/subsonicupdate.py

```diff
@@ -88,26 +90,35 @@ def __format_url():
                 context_path = ''
             url = "http://{}:{}{}".format(host, port, context_path)
 
-        return url + '/rest/startScan'
+        return url + '/rest/startScan.view'
 
     def start_scan(self):
         user = config['subsonic']['user'].as_str()
         url = self.__format_url()
-        salt, token = self.__create_token()
-
-        payload = {
-            'u': user,
-            't': token,
-            's': salt,
-            'v': '1.15.0',  # Subsonic 6.1 and newer.
-            'c': 'beets',
-            'f': 'json'
-        }
+        if config['subsonic']['user'] == 'token':
+            salt, token = self.__create_token()
+            payload = {
+                'u': user,
+                't': token,
+                's': salt,
+                'v': '1.15.0',  # Subsonic 6.1 and newer.
+                'c': 'beets',
+                'f': 'json'
+            }
+        else:
+            password = config['subsonic']['pass'].as_str()
+            encpass = hexlify(password.encode()).decode()
+            payload = {
+                'u': user,
+                'p': 'enc:{}'.format(encpass),
+                'v': '1.15.0',
+                'c': 'beets',
+                'f': 'json'
+            }
 
         try:
             response = requests.get(url, params=payload)
             json = response.json()
-
```

Comment:
Is this whitespace change intentional? Personally, I'd say this block of code is sufficiently long to benefit from the additional linebreak.

Reply:
linebreak rollback